ChatSecure/Classes/Controllers/XMPP/OTRYapDatabaseRosterStorage.m

```diff
@@ -180,6 +180,11 @@ - (void)updateBuddy:(nullable OTRXMPPBuddy *)buddy withItem:(nonnull NSXMLElemen
     [[OTRDatabaseManager sharedInstance].readWriteDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
         [newBuddy saveWithTransaction:transaction];
     }];
+    
+    if (buddy.pendingApproval && !newBuddy.pendingApproval) {
+        // Buddy was approved
+        [[NSNotificationCenter defaultCenter] postNotificationName:OTRBuddyPendingApprovalDidChangeNotification object:buddy userInfo:nil];
```

Comment:
@N-Pex I agree, that's some pretty old code over there. I think back then we ignored that the whole point of the "object" was that it acted like a "sender" parameter so you could optimize/filter out notifications from only objects you care about.